app/controllers/users_controller.rb

```diff
@@ -23,13 +25,32 @@ def edit
   end
 
   def update
-    @user = current_user
-    ConfirmationSender.send_confirmation_to(@user)
-    redirect_to user_confirmation_path
+    @user = only_current_user
+    if @user.update_attributes(user_params)
+      @user.add_role if params[:user][:role_to_add]
+      flash[:success] = "Your account information has been updated!"
+      redirect_to user_path(@user)
+    else
+      render :edit
+    end
+    # ConfirmationSender.send_confirmation_to(@user)
+    # redirect_to user_confirmation_path
   end
 
   private
     def user_params
       params.require(:user).permit(:name, :username, :status, :email, :location, :phone, :password, :password_confirmation)
     end
+
+    # below method was added to account for the edge case
+    # when users try to go to someone else's page,
+    # they can only see their own info regardless
+    # but this stops them from even trying
+    def only_current_user
+      if params[:id].to_i == current_user.id
+        current_user
+      else
```

Comment:
I tried it as a before_action and I think it is not a good idea at least now because it broke a bunch of tests/functionality since I would have to go through each and every controller and add skip it before relevant actions. Cannot do this without my teammates cursing me later on, I guess :). So let us keep it local to that controller and refactor to before_action later on. Also I would not mind deleting this method since it seems a super edge case I could not find anybody account for after searching online.